doi/src/main/java/ca/nrc/cadc/doi/DoiAction.java

```diff
@@ -205,7 +284,271 @@ private void parsePath() {
+        }
+
+    }
+
+    protected boolean checkSubjectsMatch(Subject subA, Subject subB) {
+        Set<Principal> subAPrincipals = subA.getPrincipals();
+        Set<Principal> subBPrincipals = subB.getPrincipals();
+
+        for (Principal subAPrincipal : subAPrincipals) {
+            if (subBPrincipals.contains(subAPrincipal)) {
+                // Return if one of the principals matches
+                return true;
+            }
+        }
+        return false;
+    }
+
+    protected ContainerNode getDoiContainer(String doiPath) {
+        ContainerNode doiContainer;
+        try {
+            doiContainer = (ContainerNode) vospaceDoiClient.getVOSpaceClient().getNode(doiPath);
+        } catch (IOException | InterruptedException | NodeParsingException | NodeNotSupportedException
+                 | ResourceAlreadyExistsException | ResourceNotFoundException e) {
+            throw new RuntimeException("Failed to get DOI Container from DOI Path: " + doiPath, e);
+        }
+        return doiContainer;
+    }
+
+    protected boolean isCallingUserDOIAdmin() {
+        return callingSubject != null && checkSubjectsMatch(callingSubject, getAdminSubject());
+    }
+
+    protected boolean isCallingUserPublisher() {
+        if (publisherGroupURI != null) {
+            return getGMSClient().isMember(publisherGroupURI);
+        }
+        return false;
+    }
+
+    protected boolean isCallingUserRequester() {
+        String doiPath = String.format("%s/%s", parentPath, doiSuffix);
+        ContainerNode doiContainer = getDoiContainer(doiPath);
+        Subject requestorSubject = getRequesterSubject(doiContainer);
+
+        return checkSubjectsMatch(callingSubject, requestorSubject);
+    }
+
+    protected boolean isCallingUserRequester(Node node) {
+        try {
+            Long requesterUserId = Long.parseLong(node.getProperty(DOI_VOS_REQUESTER_PROP).getValue());
+            return callersNumericId.equals(requesterUserId);
+        } catch (NumberFormatException ex) {
+            log.error(String.format("Unable to parse requester uid[%s] for doi: %s",
+                    node.getProperty(DOI_VOS_REQUESTER_PROP).getValue(), node.getName()), ex);
+            return false;
+        }
+    }
+
+    protected Subject getRequesterSubject(ContainerNode doiContainer) {
+        String doiRequester = doiContainer.getPropertyValue(DOI_VOS_REQUESTER_PROP);
+
+        if (doiRequester == null) {
+            throw new IllegalStateException("No requester associated with DOI: " + doiSuffix);
+        }
+
+        ACIdentityManager acIdentityManager = new ACIdentityManager();
+        Integer numericID = Integer.parseInt(doiRequester);
+        return acIdentityManager.toSubject(numericID);
+    }
+
+    protected List<Node> getOwnedDOIList() throws Exception {
+        List<Node> ownedNodes = new ArrayList<>();
+        ContainerNode doiRootNode = vospaceDoiClient.getContainerNode("");
```

Comment:
This should use the getDOIContainer() method with an empty path. If the parent container for all DOI's isn't found, the action can't proceed and an exception should be thrown, which getDOIContainer() does.